In [2]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load in 

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the "../input/" directory.
# For example, running this (by clicking run or pressing Shift+Enter) will list the files in the input directory

import os
print(os.listdir("../datacamp-projects/"))

# Any results you write to the current directory are saved as output.

['.git', '.ipynb_checkpoints', 'AppleStore.csv', 'datasets', 'discovery_of_handwashing.ipynb', 'draw_flowers_with_mathematics.ipynb', 'lists_loops.ipynb', 'README.md']


In [12]:
df = pd.read_csv('../datacamp-projects/AppleStore.csv')

In [13]:
df.head(5)

Unnamed: 0         id                                         track_name  \
0           1  281656475                                    PAC-MAN Premium   
1           2  281796108                          Evernote - stay organized   
2           3  281940292    WeatherBug - Local Weather, Radar, Maps, Alerts   
3           4  282614216  eBay: Best App to Buy, Sell, Save! Online Shop...   
4           5  282935706                                              Bible   

   size_bytes currency  price  rating_count_tot  rating_count_ver  \
0   100788224      USD   3.99             21292                26   
1   158578688      USD   0.00            161065                26   
2   100524032      USD   0.00            188583              2822   
3   128512000      USD   0.00            262241               649   
4    92774400      USD   0.00            985920              5320   

   user_rating  user_rating_ver     ver cont_rating   prime_genre  \
0          4.0              4.5   6.3.5          4+         Games   
1          4.0              3.5   8.2.2          4+  Productivity   
2          3.5              4.5   5.0.0          4+       Weather   
3          4.0              4.5  5.10.0         12+      Shopping   
4          4.5              5.0   7.5.1          4+     Reference   

   sup_devices.num  ipadSc_urls.num  lang.num  vpp_lic  
0               38                5        10        1  
1               37                5        23        1  
2               37                5         3        1  
3               37                5         9        1  
4               37                5        45        1

In [6]:
df.shape

(7197, 17)

In [50]:
# # opened_file = open('../input/AppleStore.csv')
# # read_file = opened_file.read()
# # read_file
# split_dataset = read_file.split('\n')
# header = split_dataset[0]
# first_5 = split_dataset[1:6]
# print(header)
# print(first_5)

"","id","track_name","size_bytes","currency","price","rating_count_tot","rating_count_ver","user_rating","user_rating_ver","ver","cont_rating","prime_genre","sup_devices.num","ipadSc_urls.num","lang.num","vpp_lic"
['"1","281656475","PAC-MAN Premium",100788224,"USD",3.99,21292,26,4,4.5,"6.3.5","4+","Games",38,5,10,1', '"2","281796108","Evernote - stay organized",158578688,"USD",0,161065,26,4,3.5,"8.2.2","4+","Productivity",37,5,23,1', '"3","281940292","WeatherBug - Local Weather, Radar, Maps, Alerts",100524032,"USD",0,188583,2822,3.5,4.5,"5.0.0","4+","Weather",37,5,3,1', '"4","282614216","eBay: Best App to Buy, Sell, Save! Online Shopping",128512000,"USD",0,262241,649,4,4.5,"5.10.0","12+","Shopping",37,5,9,1', '"5","282935706","Bible",92774400,"USD",0,985920,5320,4.5,5,"7.5.1","4+","Reference",37,5,45,1']


In [15]:
opened_file = open('../datacamp-projects/AppleStore.csv', encoding="utf8")
from csv import reader
read_file = reader(opened_file)
apps_data = list(read_file)

In [16]:
len(apps_data)

7198

In [17]:
print(apps_data[0])

['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


In [18]:
print(apps_data[1:3])

[['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1'], ['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']]


In [19]:
col_names = apps_data[0]

In [20]:
apps_data = apps_data[1:]

In [21]:
len(apps_data)

7197

In [15]:
print(col_names)

['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


In [16]:
print(apps_data[0:5])

[['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1'], ['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1'], ['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583', '2822', '3.5', '4.5', '5.0.0', '4+', 'Weather', '37', '5', '3', '1'], ['4', '282614216', 'eBay: Best App to Buy, Sell, Save! Online Shopping', '128512000', 'USD', '0', '262241', '649', '4', '4.5', '5.10.0', '12+', 'Shopping', '37', '5', '9', '1'], ['5', '282935706', 'Bible', '92774400', 'USD', '0', '985920', '5320', '4.5', '5', '7.5.1', '4+', 'Reference', '37', '5', '45', '1']]


In [17]:
row_1 = ['Facebook', 0.0, 'USD', 2974676, 3.5]
row_2 = ['Instagram', 0.0, 'USD', 2161558, 4.5]
row_3 = ['Clash of Clans', 0.0, 'USD', 2130805, 4.5]
row_4 = ['Temple Run', 0.0, 'USD', 1724546, 4.5]
row_5 = ['Pandora - Music & Radio', 0.0, 'USD', 1126879, 4.0]

app_data_set = [row_1, row_2, row_3, row_4, row_5]

for row in app_data_set:
    print(row)
    print('----------------')

# for row in app_data_set:
#     print(sum(int(row[-1])))

['Facebook', 0.0, 'USD', 2974676, 3.5]
----------------
['Instagram', 0.0, 'USD', 2161558, 4.5]
----------------
['Clash of Clans', 0.0, 'USD', 2130805, 4.5]
----------------
['Temple Run', 0.0, 'USD', 1724546, 4.5]
----------------
['Pandora - Music & Radio', 0.0, 'USD', 1126879, 4.0]
----------------


In [18]:
for element in row_1:
    print(element)
    print('---------------')

Facebook
---------------
0.0
---------------
USD
---------------
2974676
---------------
3.5
---------------


In [19]:
list_a = [10, 20, 30, 40]

In [20]:
for element in list_a:
    print(element)
    print('---------------')

10
---------------
20
---------------
30
---------------
40
---------------


In [21]:
list_of_lists = [[1,2], [11,12], [21,22]]

In [22]:
for element in list_of_lists:
    print(element[0])

1
11
21


In [23]:
for element in list_of_lists:
    print(element[-1])

2
12
22


In [24]:
for element in list_of_lists:
    print(element)

[1, 2]
[11, 12]
[21, 22]


In [25]:
a_sum = 0
list_b = [1, 3, 5]
print(a_sum)

for value in list_b:
    a_sum += value
#     a_sum = a_sum + value
    print(a_sum)

0
1
4
9


In [1]:
a_list = [1,3,5]
a_sum = 0
for value in a_list:
    a_sum += value
    print(a_sum)
    
print(a_sum)

1
4
9
9


### Find the average app rating for the apps stored in the `app_data_set` variable. 
*  Initialize a variable named rating_sum with a value of zero outside the loop body.
*  Loop (iterate) over the app_data_set list of lists. For each of the five iterations of the loop:
    *         Extract the rating of the app and store it to a variable named rating.
    *         Add the value stored in rating to the current value of the rating_sum.
*  Outside the loop body, divide the rating sum (stored in rating_sum) by the number of ratings to get an average value. Store the result in a variable named avg_rating.

In [28]:
# rating_sum = 0
# for row in app_data_set:
#     rating = row[-1]
#     rating_sum = rating_sum + rating
    
# avg_rating = rating_sum / len(app_data_set)
rating_sum = 0
for row in app_data_set:
    rating = row[-1]
    rating_sum += rating

avg_rating = rating_sum / len(app_data_set)
print(avg_rating)

4.2


### Compute the average app rating for all the 7,197 apps stored in the data set.
* Transform the AppleStore.csv file into a list of lists.
    * Open the file using the open() command. Save the output to a variable named opened_file.
    * Read in the opened file using the reader() command. Save the output to a variable named read_file. You'll have to import the reader() command from the csv module.
    * Transform the read-in file to a list of lists using the list() command. Save the list of lists to a variable named apps_data.
* Initialize a variable named rating_sum with a value of zero.
* Loop through the apps_data[1:] list of lists (make sure you don't include the header row). For each of the 7,197 iterations of the loop:
     * Extract the rating of the app and store it to a variable named rating. Make sure you convert the rating value from a string to a float.
     * Add the value stored in rating to the current value of the rating_sum.
* Divide the rating sum (stored in rating_sum) by the number of ratings to get an average value. Store the result in a variable named avg_rating.

In [30]:
rating_sum = 0
for row in apps_data:
    rating = float(row[7])
    rating_sum += rating
    
avg_rating = rating_sum / len(apps_data)
print(avg_rating)

460.3739057940809


### Using the append method on a list

In [31]:
z = [7, 4, 3, 2]
z.append(3)
print(z)

[7, 4, 3, 2, 3]


In [37]:
# 1. initialize empty list
all_ratings = []
# 2. loop over the data set and extract rating
for row in apps_data:
    rating = float(row[8])
    all_ratings.append(rating) # 3. append rating to the empty list
    
print(all_ratings)
avg_rating = sum(all_ratings) / len(all_ratings) # 4. sum up the ratings and divide the ratings by the number of ratings 
print(avg_rating)

[4.0, 4.0, 3.5, 4.0, 4.5, 4.0, 4.0, 4.0, 4.5, 4.0, 4.5, 3.5, 3.5, 3.5, 4.5, 4.0, 3.5, 4.0, 4.0, 4.0, 4.5, 5.0, 4.0, 4.5, 4.0, 3.5, 3.0, 3.5, 4.0, 4.0, 3.5, 4.5, 3.0, 3.5, 4.0, 4.0, 3.5, 4.0, 4.5, 4.0, 4.5, 3.0, 3.5, 4.5, 3.0, 3.5, 3.0, 3.0, 3.0, 4.0, 4.0, 3.0, 4.5, 3.5, 3.0, 3.5, 4.5, 4.0, 4.0, 4.0, 3.0, 3.5, 3.5, 4.0, 3.0, 4.5, 3.5, 4.5, 4.5, 3.5, 3.5, 3.5, 3.5, 4.0, 3.0, 3.5, 4.0, 3.5, 4.0, 3.5, 3.5, 4.5, 3.5, 4.0, 3.5, 4.5, 2.0, 4.5, 4.5, 4.0, 4.0, 3.5, 4.0, 3.5, 3.0, 4.0, 3.5, 2.5, 4.5, 4.5, 3.0, 4.0, 3.5, 4.5, 3.0, 4.0, 3.0, 3.5, 3.5, 4.5, 4.5, 3.0, 4.0, 4.0, 4.0, 4.0, 4.0, 3.5, 4.5, 3.0, 3.0, 4.0, 3.5, 4.0, 4.0, 4.5, 4.0, 4.5, 4.0, 4.5, 3.0, 4.5, 4.5, 4.5, 4.0, 4.0, 4.0, 4.0, 3.5, 4.5, 4.0, 3.5, 3.5, 3.0, 4.0, 4.0, 4.0, 4.0, 4.0, 3.0, 3.5, 4.0, 3.5, 4.0, 3.0, 3.5, 4.0, 4.0, 4.5, 3.5, 2.5, 3.0, 3.5, 3.5, 3.0, 3.0, 4.5, 4.0, 4.0, 4.0, 4.5, 3.5, 4.5, 4.0, 4.5, 3.5, 4.0, 3.0, 4.5, 3.5, 3.5, 4.0, 4.5, 4.0, 4.0, 3.5, 3.0, 4.5, 3.0, 3.0, 3.5, 4.5, 3.0, 4.0, 3.5, 4.0, 4.0, 3.5, 4.5, 0.0,

In [44]:
# some_string = '"1","281656475","PAC-MAN Premium",100788224,"USD",3.99,21292,26,4,4.5,"6.3.5","4+","Games",38,5,10,1\n"2","281796108","Evernote - stay organized",158578688,"USD",0,161065,26,4,3.5,"8.2.2","4+","Productivity",37,5,23,1'

In [43]:
# some_string.split('\n')

### Using the split method on a list

In [51]:
# opened_file = open('../input/AppleStore.csv')
# read_file = opened_file.read()
# read_file
split_dataset = read_file.split('\n')
header = split_dataset[0]
first_5 = split_dataset[1:6]
print(header)
print(first_5)

"","id","track_name","size_bytes","currency","price","rating_count_tot","rating_count_ver","user_rating","user_rating_ver","ver","cont_rating","prime_genre","sup_devices.num","ipadSc_urls.num","lang.num","vpp_lic"
['"1","281656475","PAC-MAN Premium",100788224,"USD",3.99,21292,26,4,4.5,"6.3.5","4+","Games",38,5,10,1', '"2","281796108","Evernote - stay organized",158578688,"USD",0,161065,26,4,3.5,"8.2.2","4+","Productivity",37,5,23,1', '"3","281940292","WeatherBug - Local Weather, Radar, Maps, Alerts",100524032,"USD",0,188583,2822,3.5,4.5,"5.0.0","4+","Weather",37,5,3,1', '"4","282614216","eBay: Best App to Buy, Sell, Save! Online Shopping",128512000,"USD",0,262241,649,4,4.5,"5.10.0","12+","Shopping",37,5,9,1', '"5","282935706","Bible",92774400,"USD",0,985920,5320,4.5,5,"7.5.1","4+","Reference",37,5,45,1']


In [54]:
# opened_file = open('AppleStore.csv')
# read_file = opened_file.read()
# split_dataset = read_file.split('\n')
final_dataset = []
for row in split_dataset:
    split_row = row.split(',')
    final_dataset.append(split_row)
    
print(final_dataset[:6])

header = final_dataset[0]
instagram = final_dataset[2]
insta_rating = instagram[7]

[['""', '"id"', '"track_name"', '"size_bytes"', '"currency"', '"price"', '"rating_count_tot"', '"rating_count_ver"', '"user_rating"', '"user_rating_ver"', '"ver"', '"cont_rating"', '"prime_genre"', '"sup_devices.num"', '"ipadSc_urls.num"', '"lang.num"', '"vpp_lic"'], ['"1"', '"281656475"', '"PAC-MAN Premium"', '100788224', '"USD"', '3.99', '21292', '26', '4', '4.5', '"6.3.5"', '"4+"', '"Games"', '38', '5', '10', '1'], ['"2"', '"281796108"', '"Evernote - stay organized"', '158578688', '"USD"', '0', '161065', '26', '4', '3.5', '"8.2.2"', '"4+"', '"Productivity"', '37', '5', '23', '1'], ['"3"', '"281940292"', '"WeatherBug - Local Weather', ' Radar', ' Maps', ' Alerts"', '100524032', '"USD"', '0', '188583', '2822', '3.5', '4.5', '"5.0.0"', '"4+"', '"Weather"', '37', '5', '3', '1'], ['"4"', '"282614216"', '"eBay: Best App to Buy', ' Sell', ' Save! Online Shopping"', '128512000', '"USD"', '0', '262241', '649', '4', '4.5', '"5.10.0"', '"12+"', '"Shopping"', '37', '5', '9', '1'], ['"5"', '"282